src/AuthenticationService.php

```diff
@@ -214,10 +215,14 @@ public function authenticate(ServerRequestInterface $request): ResultInterface
     public function clearIdentity(ServerRequestInterface $request, ResponseInterface $response): array
     {
         foreach ($this->authenticators() as $authenticator) {
-            if ($authenticator instanceof PersistenceInterface) {
+            if ($authenticator instanceof ImpersonationInterface) {
```

Comment:
Changing the interface check here means the identity won't be cleared for `CookieAuthenticator` which only implements `PersistenceInterface`, not `ImpersonationInterface`. I don't think that's expected.